### PR TITLE
Bootstrap a simple argument parser help message for end-users

### PR DESCRIPTION
**Pull request by xSavitar**
This will be expanded upon as we keep adding new flag/options to the program.

### CODE REVIEW

File: cli/main.c [hunk 1]
```diff
@@ -131,7 +131,12 @@ PkStringPtr loadScript(PKVM* vm, const char* path) {
 
 int main(int argc, char** argv) {
 
-  const char* usage = "usage: pocket [-c cmd | file]\n";
+  const char* usage =
+    "usage: pocket [-c cmd | file]\n\n"
+    "The following options are available:\n"
+    "  -c cmd | file    executes a command or a file [TODO]\n"
```
Comment (ThakeeNathees) [inline cli/main.c]: Why `TODO` here? we support running a file and an inline command in the `cli` already.

Comment (xSavitar) [reply cli/main.c]: Oh okay! Didn't yet test that. Let me remove :) Thanks

Comment (ThakeeNathees) [reply cli/main.c]: Also, the `-c cmd` and `file` are 2 different options. They should be in their own lines. Here is the python's help

```
-c cmd : program passed in as string (terminates option list)
file   : program read from script file
```

Comment (xSavitar) [reply cli/main.c]: Okay, thanks for informing me.

**Comment by xSavitar** [comment]
I think this is good now :)

File: cli/main.c [hunk 2]
```diff
@@ -131,14 +131,26 @@ PkStringPtr loadScript(PKVM* vm, const char* path) {
...
+
+  if (argc > 1) {
+    if (strcmp(argv[1], "--help") == 0) {
+      printf("%s", usage);
+      return 0;
+    } else if (strcmp(argv[1], "--version") == 0) {
+      printf("Version: %s\n", PK_VERSION_STRING);
+      return 0;
+    } else {
+      printf("Invalid option flag. Use --help for more.\n");
```
Comment (ThakeeNathees) [inline cli/main.c:150]: We can't run `pocket file.pk` anymore since `file.pk` here isn't a valid option.

Comment (ThakeeNathees) [reply cli/main.c:150]: Ooh, and the error messages should be printed to `stderr` in this case.

**Comment by ThakeeNathees** [comment]
@xSavitar Thanks for opening this PR as I'm thinking about writing an argument parser for a while. If you're interested in writing a minimal but fully function one, maybe I could open an issue, where we can discuss the implementation?

File: cli/main.c (hunk 2 above)
Comment (ThakeeNathees) [reply cli/main.c:150]: Ooh, and also error messages should be written to `stderr` in this case.

**Comment by xSavitar** [comment]
> @xSavitar Thanks for opening this PR as I'm thinking about writing an argument parser for a while. If you're interested in writing a minimal but fully function one, maybe I could open an issue, where we can discuss the implementation?

Yes! Let's discuss!

**Comment by ThakeeNathees** [comment]
Closing as it's for the future work, and once we've decided on the implementation I'll reopen this or you can create a new PR.

Anyways thanks for this PR!

**Pull request closed without merging**